code/game/machinery/status_display.dm

```diff
@@ -18,49 +14,34 @@
 	density = FALSE
 	use_power = IDLE_POWER_USE
 	idle_power_usage = 10
-	var/mode = 1	// 0 = Blank
-					// 1 = Shuttle timer
-					// 2 = Arbitrary message(s)
-					// 3 = alert picture
-					// 4 = Station time
-
-	var/picture_state	// icon_state of alert picture
-	var/message1 = ""	// message line 1
-	var/message2 = ""	// message line 2
-	var/index1			// display index for scrolling messages or 0 if non-scrolling
-	var/index2
-
-	frequency = DISPLAY_FREQ		// radio frequency
-
-	var/friendc = FALSE      // track if Friend Computer mode
-	var/ignore_friendc = FALSE
-
-	var/spookymode = FALSE
-
 	maptext_height = 26
 	maptext_width = 32
 	maptext_y = -1
 
-	var/const/CHARS_PER_LINE = 5
-	var/const/STATUS_DISPLAY_BLANK = 0
-	var/const/STATUS_DISPLAY_TRANSFER_SHUTTLE_TIME = 1
-	var/const/STATUS_DISPLAY_MESSAGE = 2
-	var/const/STATUS_DISPLAY_ALERT = 3
-	var/const/STATUS_DISPLAY_TIME = 4
-	var/const/STATUS_DISPLAY_CUSTOM = 99
+	/// Status display mode
+	VAR_PRIVATE/mode = STATUS_DISPLAY_TRANSFER_SHUTTLE_TIME
+	/// icon_state of alert picture
+	var/picture_state
+	/// Are we spooked?
+	var/spookymode = FALSE
+	/// Line 1 of a custom message, if any
+	var/message1
+	/// Line 2 of a custom message, if any
+	var/message2
+	/// Is this a supply display?
+	var/is_supply = FALSE
```

Comment:
shouldn't this just be a subtype? (for future reference that's all, otherwise this would be a map edit)

Reply:
The supply subtype defaults this to true.

Dont quote me on this, but I am pretty sure `GetVar()` instruction calls are cheaper than `istype()`.